File: FrontEnd/app_loteria/lib/models/Venta.dart

class VentaJsonDetalle {
  int? numeroId;
  int? valor;

  VentaJsonDetalle({
    this.numeroId,
    this.valor,
  });

  Map<String, dynamic> toJson() {
    return {
      "numeroId": numeroId,
      "valor": valor,
    };
  }
}

class VentaJsonEncabezado {
  int? personaId;
  int? usuarioId;
  int? metodoPagoId;
  String fechaVenta;
  int? usuarioCreacion;
  List<VentaJsonDetalle> ventaDetalles;

  VentaJsonEncabezado({
    this.personaId,
    this.usuarioId,
    this.metodoPagoId,
    required this.fechaVenta,
    this.usuarioCreacion,
    required this.ventaDetalles,
  });

  Map<String, dynamic> toJson() {
    return {
      "personaId": personaId,
      "usuarioId": usuarioId,
      "metodoPagoId": metodoPagoId,
      "fechaVenta": fechaVenta,
      "usuarioCreacion": usuarioCreacion,
      "ventaDetalles": ventaDetalles.map((detalle) => detalle.toJson()).toList(),
    };
  }
}

File: Backend/Numerito.API/Numerito.API/Data/Entities/Venta.cs
using FluentValidation;

namespace Numerito.API.Data.Entities;

public partial class Venta
{
    public int VentaId { get; set; }

    public int PersonaId { get; set; }

    public int UsuarioId { get; set; }

    public int MetodoPagoId { get; set; }

    public DateTime FechaVenta { get; set; }

    public int UsuarioCreacion { get; set; }

    public DateTime FechaCreacion { get; set; }

    public int? UsuarioModificacion { get; set; }

    public DateTime? FechaModificacion { get; set; }

    public bool? Estado { get; set; }

    public virtual MetodosPago MetodoPago { get; set; } = null!;

    public virtual Persona Persona { get; set; } = null!;

    public virtual Usuario Usuario { get; set; } = null!;

    public virtual Usuario UsuarioCreacionNavigation { get; set; } = null!;

    public virtual Usuario? UsuarioModificacionNavigation { get; set; }

    public virtual ICollection<VentaDetalle> VentaDetalles { get; set; } = new List<VentaDetalle>();

    public class VentaValidations : AbstractValidator<Venta>
    {
        public VentaValidations()
        {
            RuleFor(x => x.PersonaId).NotEmpty();
            RuleFor(x => x.UsuarioId).NotEmpty();
            RuleFor(x => x.MetodoPagoId).NotEmpty();
        }
    }
}
